### ISSUE TITLE
hiredis adapters libuv error for help

### ISSUE THREAD

**Issue by zhjzjnb** [redis/hiredis#508]
Hello everyone.
I'm trying to use libuv to connect redis with hiredis,but there are some differences between macos 10.11.3 and ubuntu 12.04 LTS,if my redis-server is down,when I try to connect redis,the 'ConnectCallback' will work OK in macos,it returns 'connectCallback Error: Connection refused', however in ubuntu it returns nothing.I see the hiredis-libuv-adapters,here is a function 'static void redisLibuvPoll(uv_poll_t* handle, int status, int events)',the 'status' is not 0 in ubuntu, and I can't find why.
ask for help.thanks.
my libuv version is v1.11.0: 2017.02.02, Version 1.11.0 (Stable)
hiredis version is v0.13.3(github:https://github.com/redis/hiredis)

**Comment by badboy** [comment]
Can you provide a minimal example experiencing this issue?

**Comment by zhjzjnb** [comment]
this is my demo.
https://github.com/zhjzjnb/myredis

**Comment by badboy** [comment]
Looks like `status` on that call is `UV_EBADF` (`-9`). Likely because at that point the file descriptor is already invalid.
However, that never reaches the hiredis code.

One hotfix that seems to work for this case is forcing a write event on error status:

```patch
diff --git i/adapters/libuv.h w/adapters/libuv.h
index 3cdf3d3..c8a6d30 100644
--- i/adapters/libuv.h
+++ w/adapters/libuv.h
@@ -17,7 +17,7 @@ static void redisLibuvPoll(uv_poll_t* handle, int status, int events) {
   redisLibuvEvents* p = (redisLibuvEvents*)handle->data;
 
   if (status != 0) {
-    return;
+    events |= UV_WRITABLE;
   }
 
   if (events & UV_READABLE) {
```

This feels hacky but I do not know a better way right now.

**Comment by badboy** [comment]
Oh wait, here is another PR tackling this: https://github.com/redis/hiredis/pull/461

**Comment by michael-grunder** [comment]
Fixed in #578
